sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/ComplexTypes.scala

```diff
@@ -47,7 +47,8 @@ object SimplifyExtractValueOps extends Rule[LogicalPlan] {
           // For example, if a user submits a query like this:
           // `$"struct_col".withField("b", lit(1)).withField("b", lit(2)).getField("b")`
           // we want to return `lit(2)` (and not `lit(1)`).
-          matches.last._2
+          val expr = matches.last._2
+          If(IsNull(struct), Literal(null, expr.dataType), expr)
```

Comment:
I think we need this only if `struct.nullable` is true?

Reply:
other optimizer rules will optimize it, but yea it's better to not introduce unnecessary plan change.